pkgs/by-name/ke/keepassxc/package.nix

```
@@ -77,18 +77,36 @@ stdenv.mkDerivation (finalAttrs: {
     ];
 
   doCheck = true;
-  checkPhase = ''
-    runHook preCheck
-
-    export LC_ALL="en_US.UTF-8"
-    export QT_QPA_PLATFORM=offscreen
-    export QT_PLUGIN_PATH="${libsForQt5.qtbase.bin}/${libsForQt5.qtbase.qtPluginPrefix}"
-    # testcli, testgui and testkdbx4 are flaky - skip them all
-    # testautotype on darwin throws "QWidget: Cannot create a QWidget without QApplication"
-    make test ARGS+="-E 'testcli|testgui${lib.optionalString stdenv.hostPlatform.isDarwin "|testautotype|testkdbx4"}' --output-on-failure"
-
-    runHook postCheck
-  '';
+  checkPhase =
+    ''
+      runHook preCheck
+
+      export LC_ALL="en_US.UTF-8"
+      export QT_QPA_PLATFORM=offscreen
+      export QT_PLUGIN_PATH="${libsForQt5.qtbase.bin}/${libsForQt5.qtbase.qtPluginPrefix}"
+    ''
+    + (
+      let
+        disabledTests = lib.concatStringsSep "|" (
+          [
+            # flaky
+            "testcli"
+            "testgui"
+          ]
+          ++ lib.optionals stdenv.hostPlatform.isDarwin [
+            # QWidget: Cannot create a QWidget without QApplication
+            "testautotype"
+
+            "testkdbx4"
+          ]
+        );
+      in
+      ''
+        make test ARGS+="-E '${disabledTests}' --output-on-failure"
+
+        runHook postCheck
+      ''
+    );
```

I'd use a `let ... in` block _before_ the string, rather than in the middle of it.

```suggestion
    let
      disabledTests = [
        # flaky
        "testcli"
        "testgui"
      ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
        # QWidget: Cannot create a QWidget without QApplication
        "testautotype"
        "testkdbx4"
      ];
    in
    ''
      runHook preCheck

      export LC_ALL="en_US.UTF-8"
      export QT_QPA_PLATFORM=offscreen
      export QT_PLUGIN_PATH="${libsForQt5.qtbase.bin}/${libsForQt5.qtbase.qtPluginPrefix}"
      make test ARGS+="-E '${lib.concatStringsSep "|" disabledTests}' --output-on-failure"

      runHook postCheck
    ''